### PR TITLE
Adding ProtBERT 

### PR DESCRIPTION
**Pull request by Shiva-sankaran**
## Description

Fix #(issue)

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [x] Run `mypy -p deepchem` and check no errors
  - [x] Run `flake8 <modified file> --count` and check no errors
  - [x] Run `python -m doctest <modified file>` and check no errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: deepchem/models/torch_models/prot_bert.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    """
+
+    def __init__(self,
+                 task: str,
+                 model_path: str = 'Rostlab/prot_bert',
+                 n_tasks: int = 1,
+                 cls_name: str = "LogReg",
+                 **kwargs):
+        self.n_tasks = n_tasks
+        tokenizer = BertTokenizer.from_pretrained(model_path,
```
Comment (NimishaDey) [inline deepchem/models/torch_models/prot_bert.py:78]: Add type annotations

File: deepchem/models/torch_models/prot_bert.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    .. [2] https://huggingface.co/Rostlab/prot_bert
+
+    """
+
+    def __init__(self,
+                 task: str,
+                 model_path: str = 'Rostlab/prot_bert',
+                 n_tasks: int = 1,
+                 cls_name: str = "LogReg",
+                 **kwargs):
```
Comment (NimishaDey) [inline deepchem/models/torch_models/prot_bert.py:76]: Add docstrings explaining the parameters

File: deepchem/models/torch_models/prot_bert.py (hunk 1 above)
Comment (Shiva-sankaran) [reply deepchem/models/torch_models/prot_bert.py:78]: Added.

File: deepchem/models/torch_models/prot_bert.py (hunk 2 above)
Comment (Shiva-sankaran) [reply deepchem/models/torch_models/prot_bert.py]: Added.

File: deepchem/models/torch_models/prot_bert.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
+from transformers import BertForMaskedLM, BertForSequenceClassification, BertTokenizer, BertConfig
+import torch.nn as nn
+from deepchem.models.torch_models.hf_models import HuggingFaceModel
+from typing import Union
+
+
+class ProtBERT(HuggingFaceModel):
```
[the remark this answers is not in the document]
Comment (Shiva-sankaran) [reply deepchem/models/torch_models/prot_bert.py:7]: image
image

File: deepchem/models/torch_models/tests/test_prot_bert.py [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+except ModuleNotFoundError:
+    pass
+
+
+@pytest.mark.torch
+def test_prot_bert_pretraining_mlm(protein_classification_dataset):
+    model_path = 'Rostlab/prot_bert'
+    model = ProtBERT(task='mlm', HG_model_path=model_path, n_tasks=1)
+    loss = model.fit(protein_classification_dataset, nb_epoch=1)
+    assert loss
```
Comment (NimishaDey) [inline deepchem/models/torch_models/tests/test_prot_bert.py:18]: What is the range of loss values you are getting?

Comment (Shiva-sankaran) [reply deepchem/models/torch_models/tests/test_prot_bert.py:18]: Around 0.4

**Comment by NimishaDey** [comment]
LGTM

**Review by rbharath — COMMENTED** [review]
LGTM once rebased and CI clear

@shreyasvinaya Please do CI check on this after rebase

File: deepchem/models/torch_models/prot_bert.py [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    """
+    ProtBERT model[1].
+
+    ProtBERT model is based on BERT architecture and the current implementation
```
Comment (GreatRSingh) [inline deepchem/models/torch_models/prot_bert.py:11]: @Shiva-sankaran I think a larger description of the model (ProtBert) will be better.
If anyone using this model can just read it and understand how the model works internally will be very useful.

Comment (Shiva-sankaran) [reply deepchem/models/torch_models/prot_bert.py:11]: Added.

**Review by rbharath — APPROVED** [review]
LGTM
